release/serve_tests/workloads/anyscale_service_utils.py

```diff
@@ -0,0 +1,113 @@
+import boto3
+from contextlib import contextmanager
+import logging
+import os
+from typing import Dict, List, Optional, Union
+
+from anyscale.authenticate import AuthenticationBlock
+from anyscale import AnyscaleSDK, service
+from ray._private.test_utils import wait_for_condition
+
+
+logger = logging.getLogger(__file__)
+logging.basicConfig(level=logging.INFO)
+
+
+class DeferredEnvVar:
+    def __init__(self, var: str, default: Optional[str] = None):
+        self._var = var
+        self._default = default
+
+    def __str__(self):
+        return os.environ.get(self._var, self._default)
+
+
+RELEASE_AWS_ANYSCALE_SECRET_ARN = DeferredEnvVar(
+    "RELEASE_AWS_ANYSCALE_SECRET_ARN",
+    "arn:aws:secretsmanager:us-west-2:029272617770:secret:"
+    "release-automation/"
+    "anyscale-token20210505220406333800000001-BcUuKB",
+)
+
+
+def get_anyscale_cli_token() -> str:
+    try:
+        token, _ = AuthenticationBlock._load_credentials()
+        logger.info("Loaded anyscale credentials from local storage.")
+        return token
+    except Exception:
+        pass  # Ignore errors
+
+    logger.info("Missing ANYSCALE_CLI_TOKEN, retrieving from AWS secrets store")
+    return boto3.client("secretsmanager", region_name="us-west-2").get_secret_value(
+        SecretId=str(RELEASE_AWS_ANYSCALE_SECRET_ARN)
+    )["SecretString"]
+
+
+def get_current_build_id(sdk: AnyscaleSDK) -> str:
+    cluster_id = os.environ["ANYSCALE_CLUSTER_ID"]
+    cluster = sdk.get_cluster(cluster_id)
+    return cluster.result.cluster_environment_build_id
+
+
+def build_id_to_image_uri(sdk: AnyscaleSDK, build_id: str) -> str:
+    build = sdk.get_cluster_environment_build(build_id).result
+    cluster_env = sdk.get_cluster_environment(build.cluster_environment_id).result
+    return f"anyscale/image/{cluster_env.name}:{build.revision}"
```

Comment:
let's avoid using the old `SDK` at all (assume they're not public APIs) -- why do we need to get the build_id from current cluster at all

Reply:
A BYOD image is built for release test jobs that contains the ray compiled at that commit - I want to reuse that for the service cluster.

Reply:
as discussed offline i think we should take this as an env variable

that way the test can be run from local as well

Reply:
Makes sense, I've addressed this. The cluster env name is now passed in as an env variable.